students/ejackie/session03/slicing_lab.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+
+##################################################
+## Author: Jackie Cheung
+## Date: 2018/10/23
+## Version: 0.1
+## Class: Au2018-Py210B
+## Description: Slicing Lab - Get the basics of sequence slicing down
+##################################################
+
+def exchange_first_last(seq):
+    """ with the first and last items exchanged """
+
+    first = seq[:1]
+    last = seq[-1:]
+    mid = seq[1:-1]
+    a_new_sequence = last + mid + first
+
+    return a_new_sequence
+
+def remove_first_4_last_4_every_other_remain(seq):
+    """ with the first 4 and the last 4 items removed, and then every other item in the remaining sequence """
+
+    a_new_sequence = seq[4:-4:2]
+
+    return a_new_sequence
+
+def reverse_element(seq):
+    """ with the elements reversed (just with slicing) """
+
+    a_new_sequence = seq[::-1]
+
+    return a_new_sequence
+
+def new_order_last_third_first_third_middle_third(seq):
+    """ with the last third, then first third, then the middle third in the new order """
+
+    # number of items for a third of the sequence
+    num_third = int(len(seq)/3)
```

Comment:
Minor: `len(seq)//3` may be preferred (over `int(len(seq)/3)`). Note that it's a Python 3 change. In Python 2, `3/2 == 1`, but in Python 3, `3/2 == 1.5`, and `3//2 == 1`.
```
hosungs@DESKTOP-GJT06Q0:/mnt/c/gitroot/Au2018-Py210B (master)$ python
Python 2.7.15rc1 (default, Apr 15 2018, 21:51:34)
[GCC 7.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> 3/2
1
>>>
hosungs@DESKTOP-GJT06Q0:/mnt/c/gitroot/Au2018-Py210B (master)$ python3
Python 3.6.6 (default, Sep 12 2018, 18:26:19)
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> 3/2
1.5
>>> 3//2
1
>>>
hosungs@DESKTOP-GJT06Q0:/mnt/c/gitroot/Au2018-Py210B (master)$
```

Reply:
Thanks for the suggestion! Updated the code with len(seq)//3.